irc/handlers.go

```diff
@@ -1938,6 +1938,12 @@ func dispatchMessageToTarget(client *Client, tags map[string]string, histType hi
 			}
 			return
 		}
+
+		// Restrict CTCP message for target user with +T
+		if message.IsRestrictedCTCPMessage() && user.modes.HasMode(modes.UserNoCTCP) {
```

Comment:
Optional performance-related feedback: either reverse the order of these checks, or cache the result of `IsRestrictedCTCPMessage` for use lower down with Tor sessions.

Reply:
:+1: